filament/src/PostProcessManager.cpp

```diff
@@ -281,94 +282,80 @@ FrameGraphId<FrameGraphTexture> PostProcessManager::structure(FrameGraph& fg,
     // structure pass -- automatically culled if not used, currently used by:
     //    - ssao
     //    - contact shadows
-    //    - depth-of-field
     // It consists of a mipmapped depth pass, tuned for SSAO
     struct StructurePassData {
         FrameGraphId<FrameGraphTexture> depth;
-        FrameGraphRenderTargetHandle rt;
     };
 
     // sanitize a bit the user provided scaling factor
     width  = std::max(32u, (uint32_t)std::ceil(width * scale));
     height = std::max(32u, (uint32_t)std::ceil(height * scale));
 
     // We limit the lowest lod size to 32 pixels (which is where the -5 comes from)
-    const size_t levelCount = FTexture::maxLevelCount(width, height) - 5;
+    const size_t levelCount = std::min(8, FTexture::maxLevelCount(width, height) - 5);
     assert_invariant(levelCount >= 1);
 
     // generate depth pass at the requested resolution
     auto& structurePass = fg.addPass<StructurePassData>("Structure Pass",
             [&](FrameGraph::Builder& builder, auto& data) {
-                data.depth = builder.createTexture("Structure Buffer", {
+                data.depth = builder.create<FrameGraphTexture>("Structure Buffer", {
```

Comment:
Why not use `builder.createTexture` here?

Reply:
That's an oversight. Thanks!